### PR TITLE
Feature #636 - Changes Proposals' Add to Replace

### LINKED ISSUE #636: Change addProposalsAction -> replaceProposalsAction
`addProposalsAction` adds proposals to the list. However, since we only use this on empty lists, we can change the name and functionality to `replaceProposalsAction`. `proposalsReducer` must be udated to do a replace logic instead of add.

The update code

```ts
  const loadProposals = async (): Promise<void> => {
    const governor = store.getState().extension.governor;
    if (!governor) throw new Error("Governor not set in store. This is a bug.");
    const chainProposals = await getProposals(governor);
    dispatch(replaceProposalsAction(chainProposals));
  };
```

can then be called multiple times in order to reload the list of proposals

### PR DESCRIPTION
**Pull request by abefernan**
Closes #636.

### CODE REVIEW

File: packages/sil-governance/src/routes/create-proposal/index.tsx [hunk 1]
```diff
@@ -14,10 +15,12 @@ import ProposalForm from "./components/ProposalForm";
 const CreateProposal = (): JSX.Element => {
   const dispatch = useDispatch();
   const lastSignAndPostResult = useSelector((state: RootState) => state.transactions.lastSignAndPostResult);
+  const proposals = useSelector((state: RootState) => state.proposals);
 
   const onReturnToDashboard = (): void => {
     dispatch(setTransactionsStateAction());
     history.push(DASHBOARD_ROUTE);
+    dispatch(replaceProposalsAction(proposals));
```
Comment (webmaster128) [inline packages/sil-governance/src/routes/create-proposal/index.tsx]: Why this change? Here you take data from the store and write it to the store.

To reload from chain, we need something like

```ts
  const loadProposals = async (): Promise<void> => {
    const governor = store.getState().extension.governor;
    if (!governor) throw new Error("Governor not set in store. This is a bug.");
    const chainProposals = await getProposals(governor);
    dispatch(replaceProposalsAction(chainProposals));
  };
```

File: packages/sil-governance/src/routes/dashboard/index.tsx [hunk 2]
```diff
@@ -14,10 +15,12 @@ import ProposalsList from "./components/ProposalsList";
 const Dashboard = (): JSX.Element => {
   const dispatch = useDispatch();
   const lastSignAndPostResult = useSelector((state: RootState) => state.transactions.lastSignAndPostResult);
+  const proposals = useSelector((state: RootState) => state.proposals);
 
   const onReturnToDashboard = (): void => {
     dispatch(setTransactionsStateAction());
     history.push(DASHBOARD_ROUTE);
+    dispatch(replaceProposalsAction(proposals));
```
Comment (webmaster128) [inline packages/sil-governance/src/routes/dashboard/index.tsx]: as above

File: packages/sil-governance/src/routes/create-proposal/index.tsx (hunk 1 above)
Comment (abefernan) [reply packages/sil-governance/src/routes/create-proposal/index.tsx]: Sure, I messed up there because I began to follow the suggested types... The issue I'm having is `governor.getProposals()` returns `Proposal[]` and `replaceProposalsAction` expects `ProposalProps[]`.

Comment (webmaster128) [reply packages/sil-governance/src/routes/create-proposal/index.tsx]: When you get the transaction ID, it was just posted into the network. I.e. it is not yet included in a block. This is the reason I suggetsed a 5 second delay between getting the result and triggering a reload

Comment (abefernan) [reply packages/sil-governance/src/routes/create-proposal/index.tsx]: The delay should be before showing the confirmation page, or before redirecting to Dashboard after confirming the return?

Comment (webmaster128) [reply packages/sil-governance/src/routes/create-proposal/index.tsx]: The best thing would be to have the reload independent of the screen navigation, like 5 seconds after you received the transaction ID (triggered by `setTimeout`)

File: packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx [hunk 3]
```diff
@@ -142,6 +143,16 @@ const ProposalForm = (): JSX.Element => {
...
     const transactionId = await sendSignAndPostRequest(connection, createProposalTx);
+
+    const updateChainProposals = async (): Promise<void> => {
+      const chainProposals = await getProposals(governor);
+      dispatch(replaceProposalsAction(chainProposals));
+    };
+
+    setTimeout(() => {
+      updateChainProposals();
+    }, 5000);
```
Comment (webmaster128) [inline packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx]: Suggestion: this can be written as a on-liner:

```ts
setTimeout(() => updateChainProposals(), 5000);
```

File: packages/sil-governance/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx [hunk 4]
```diff
@@ -37,6 +38,16 @@ const Buttons = ({ id, vote }: Props): JSX.Element => {
...
       const transactionId = await sendSignAndPostRequest(connection, voteTx);
+
+      const updateChainProposals = async (): Promise<void> => {
+        const chainProposals = await getProposals(governor);
+        dispatch(replaceProposalsAction(chainProposals));
+      };
+
+      setTimeout(() => {
+        updateChainProposals();
+      }, 5000);
```
Comment (webmaster128) [inline packages/sil-governance/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx]: same here

File: packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx (hunk 3 above)
Comment (abefernan) [reply packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx]: Sure! There were more lines before
